pallets/asset-index/src/lib.rs

```diff
@@ -121,6 +149,63 @@ pub mod pallet {
             Self::deposit_event(Event::AssetAdded(asset_id, units, caller, value));
             Ok(().into())
         }
+
+        /// Initiate a transfer from the user's sovereign account into the index.
+        ///
+        /// This will withdraw the given amount from the user's sovereign account and mints PINT proportionally using the latest available price pairs
+        #[pallet::weight(10_000)] // TODO: Set weights
+        pub fn deposit(
+            origin: OriginFor<T>,
+            asset_id: T::AssetId,
+            amount: BalanceFor<T>,
+        ) -> DispatchResultWithPostInfo {
+            let caller = ensure_signed(origin)?;
+
+            let mut holding = Holdings::<T>::get(&asset_id)
+                .filter(|holding| matches!(holding.availability, AssetAvailability::Liquid(_)))
+                .ok_or(Error::<T>::UnsupportedAsset)?;
+
+            let price = T::PriceFeed::get_price(asset_id.clone())?;
```

Comment:
Consider adding a test for the failure cases:
- When a price feed isn't available
- When the various units overflow
- Interacting with the MultiAssetDepository fails